apps/src/dance/DanceVisualizationColumn.jsx

```diff
@@ -21,67 +20,60 @@ const styles = {
   }
 };
 
-//TODO: Remove this during clean-up
-var songs = {
-  macklemore: {
-    url: 'https://curriculum.code.org/media/uploads/chu.mp3',
-    bpm: 146,
-    delay: 0.2, // Seconds to delay before calculating measures
-    verse: [26.5, 118.56], // Array of timestamps in seconds where verses occur
-    chorus: [92.25, 158] // Array of timestamps in seconds where choruses occur
-  },
-  macklemore90: {
-    url: 'https://curriculum.code.org/media/uploads/hold.mp3',
-    bpm: 146,
-    delay: 0.0, // Seconds to delay before calculating measures
-    verse: [0, 26.3], // Array of timestamps in seconds where verses occur
-    chorus: [65.75] // Array of timestamps in seconds where choruses occur
-  },
-  hammer: {
-    url: 'https://curriculum.code.org/media/uploads/touch.mp3',
-    bpm: 133,
-    delay: 2.32, // Seconds to delay before calculating measures
-    verse: [1.5, 15.2], // Array of timestamps in seconds where verses occur
-    chorus: [5.5, 22.1] // Array of timestamps in seconds where choruses occur
-  },
-  peas: {
-    url: 'https://curriculum.code.org/media/uploads/feeling.mp3',
-    bpm: 128,
-    delay: 0.0, // Seconds to delay before calculating measures
-    verse: [1.5, 15.2], // Array of timestamps in seconds where verses occur
-    chorus: [5.5, 22.1] // Array of timestamps in seconds where choruses occur
-  }
-};
-
 const SongSelector = Radium(class extends React.Component {
   static propTypes = {
     setSong: PropTypes.func.isRequired,
     selectedSong: PropTypes.string.isRequired,
     isProjectLevel: PropTypes.bool.isRequired
   };
 
+  state = {
+    songsData: []
+  };
+
   changeSong = (event) => {
     const song = event.target.value;
     this.props.setSong(song);
+    this.loadSong(song);
+  };
 
+  loadSong(song) {
     //Load song
     let options = {id: song};
-    options['mp3'] = songs[options.id].url;
+    options['mp3'] = `https://curriculum.code.org/media/uploads/${this.state.songsData[options.id].url}.mp3`;
     Sounds.getSingleton().register(options);
 
     if (this.props.isProjectLevel) {
       //Save song to project
       project.saveSelectedSong(song);
     }
-  };
+  }
+
+  componentDidMount() {
+    fetch(`/api/v1/sound-library/hoc_song_meta/songManifest.json`)
+      .then((response) => {return response.json();})
+      .then((data) => {this.parseSongOptions(data);});
+  }
+
+  parseSongOptions(data) {
+    let songs = {};
+    if (data) {
+      data.songs.forEach((song) => {
+          songs[song.id] = {title: song.text, url: song.url};
+      });
+    }
+    this.setState({songsData: songs});
+
+    this.loadSong(this.props.selectedSong);
```

Comment:
`setState` is not guaranteed to be synchronous; if `loadSong` depends on it, it should happen in a callback:
```js
this.setState({songsData: songs}, () => {
  this.loadSong(this.props.selectedSong);
});
```